src/core/Makefile.am

```diff
@@ -163,6 +165,8 @@ noinst_HEADERS                      = \
     thread/network_data_local.hpp     \
     thread/network_data_tlvs.hpp      \
     thread/panid_query_server.hpp     \
+    thread/network_diagnostic.hpp     \
+    thread/network_diagnostic_tlvs.hpp \
```

Comment:
nit: should be alphabetical
nit: align backspaces (you can also shorten the filename to `network_diag_*`.